SimCalorimetry/HGCalSimProducers/src/HGCDigitizerBase.cc

```diff
@@ -51,6 +51,18 @@ HGCDigitizerBase<DFr>::HGCDigitizerBase(const edm::ParameterSet& ps) : scaleByDo
   edm::ParameterSet feCfg = myCfg_.getParameter<edm::ParameterSet>("feCfg");
   myFEelectronics_ = std::unique_ptr<HGCFEElectronics<DFr>>(new HGCFEElectronics<DFr>(feCfg));
   myFEelectronics_->SetNoiseValues(noise_fC_);
+  RandNoiseGenerationFlag_ = 0;
+}
+
+template <class DFr>
+void HGCDigitizerBase<DFr>::GenerateGaussianNoise(CLHEP::HepRandomEngine* engine,
+                                                  const double NoiseMean,
+                                                  const double NoiseStd) {
+  for (size_t i = 0; i < NoiseArrayLength_; i++) {
+    for (size_t j = 0; j < samplesize_; j++) {
+      GaussianNoiseArray_[i][j] = CLHEP::RandGaussQ::shoot(NoiseMean, NoiseStd);
```

Comment:
here you need to use `CLHEP::RandGaussQ::shoot(engine, NoiseMean, NoiseStd);`, otherwise the engine is not actually used.